src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -719,7 +744,30 @@ public void performAnticompaction(ColumnFamilyStore cfs,
             txn.close();
         }
 
-        logger.info("{} Completed anticompaction successfully", PreviewKind.NONE.logPrefix(parentRepairSession));
+        logger.info("{} Completed anticompaction successfully", PreviewKind.NONE.logPrefix(sessionID));
+    }
+
+    static void validateSSTableBoundsForAnticompaction(UUID sessionID,
+                                                       Collection<SSTableReader> sstables,
+                                                       Collection<Range<Token>> fullRanges,
+                                                       Collection<Range<Token>> transRanges)
+    {
+        Set<Range<Token>> allRanges = Sets.newHashSet(Iterables.concat(fullRanges, transRanges));
+        List<Range<Token>> normalizedRanges = Range.normalize(allRanges);
+        for (SSTableReader sstable : sstables)
+        {
+            Bounds<Token> bounds = new Bounds<>(sstable.first.getToken(), sstable.last.getToken());
+
+            if (!Iterables.any(normalizedRanges, r -> (r.contains(bounds.left) && r.contains(bounds.right)) || r.intersects(bounds)))
```

Comment:
Is contains redundant with intersects?

Reply:
no they do slightly different things. For instance if the sstable range covered a superset of the range we're comparing it against, it's right and left tokens won't be contained, but the range will intersect

Reply:
It's possible, but the `intersects` logic is not totally trivial so I'm not 100%. In any case, I'd prefer to limit the logic changes in this patch to just what's required for adding support transient replication so I don't introduce any unnecessary bugs.